lib/handler/file.c

```diff
@@ -110,6 +127,57 @@ static void do_proceed(h2o_generator_t *_self, h2o_req_t *req)
         do_close(&self->super, req);
 }
 
+static void do_multirange_proceed(h2o_generator_t *_self, h2o_req_t *req) {
+    struct st_h2o_sendfile_generator_t *self = (void *)_self;
+    size_t rlen, *range_cur = self->ranged.range_infos + 2 * self->ranged.finished_range, used_buf = 0;
+    ssize_t rret, vecarrsize;
+    h2o_iovec_t vec, *vecarr;
+    int is_finished;
+
+    if (self->ranged.current_range == self->ranged.finished_range) {
```

Comment:
This is kind of a nitpick, but I suggest to make the code simpler by doing the following:
* instead of adjusting the value of `*range_cur`, use `self->bytesleft` to store the number of remaining bytes to be sent within the range, and use `self->bytesleft == 0` to detect if it is necessary to emit the boundary
 * this has a nice side-effect that the variable is used the same way as with non-multirange responses